ember-core/shared/src/main/scala/org/http4s/ember/core/Encoder.scala

```diff
@@ -139,7 +139,12 @@ private[ember] object Encoder {
           }
         }
 
-        if (!chunked && !appliedContentLength && !NoPayloadMethods.contains(req.method)) {
+        if (
+          !appliedContentLength && req.body == EmptyBody && !NoPayloadMethods.contains(req.method)
```

Comment:
It'd be good to define `def`s for those boolean expressions for better readability.

Reply:
I agree about readability but I feel superstitious about performance. Probably it will be inlined by JIT 🤔

Reply:
It will be nice when in scala 3 we write `inline` and sleep easy.